sdk/src/validation_results.rs

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    pub fn add_success_val(mut self, sm: ValidationStatus) -> Self {
+        self.success.push(sm);
+        self
+    }
+
+    pub fn success(&self) -> &Vec<ValidationStatus> {
+        self.success.as_ref()
+    }
+
+    pub fn add_informational_val(mut self, sm: ValidationStatus) -> Self {
+        self.informational.push(sm);
+        self
+    }
+
+    pub fn informational(&self) -> &Vec<ValidationStatus> {
+        self.informational.as_ref()
+    }
+
+    pub fn add_failure_val(mut self, sm: ValidationStatus) -> Self {
+        self.failure.push(sm);
+        self
+    }
+
+    pub fn failure(&self) -> &Vec<ValidationStatus> {
+        self.failure.as_ref()
+    }
+}
+
+#[derive(Clone, Serialize, Default, Deserialize, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "json_schema", derive(JsonSchema))]
+/// A map of validation results for a manifest store.
+///
+/// The map contains the validation results for the active manifest and any ingredient deltas.
+/// It is normal for there to be many
+pub struct ValidationResults {
+    #[serde(rename = "activeManifest", skip_serializing_if = "Option::is_none")]
+    active_manifest: Option<StatusCodes>, // Validation status codes for the ingredient's active manifest. Present if ingredient is a C2PA asset. Not present if the ingredient is not a C2PA asset.
+
+    #[serde(rename = "ingredientDeltas", skip_serializing_if = "Option::is_none")]
+    ingredient_deltas: Option<Vec<IngredientDeltaValidationResult>>, // List of any changes/deltas between the current and previous validation results for each ingredient's manifest. Present if the the ingredient is a C2PA asset.
+}
+
+impl ValidationResults {
+    /// Returns the [ValidationState] of the manifest store based on the validation results.
+    pub fn validation_state(&self) -> ValidationState {
+        let mut is_trusted = true; // Assume the state is trusted until proven otherwise
+        if let Some(active_manifest) = self.active_manifest.as_ref() {
+            if !active_manifest.failure().is_empty() {
+                return ValidationState::Invalid;
+            }
+            // There must be a trusted credential in the active manifest for the state to be trusted
+            is_trusted = active_manifest.success().iter().any(|status| {
+                status.code() == crate::validation_status::SIGNING_CREDENTIAL_TRUSTED
+            });
+        }
+        if let Some(ingredient_deltas) = self.ingredient_deltas.as_ref() {
+            for idv in ingredient_deltas.iter() {
+                if !idv.validation_deltas().failure().is_empty() {
+                    return ValidationState::Invalid;
+                }
+            }
+        }
+        if is_trusted {
+            ValidationState::Trusted
+        } else {
+            ValidationState::Valid
+        }
+    }
+
+    /// Returns a list of all validation errors in [ValidationResults].
+    pub fn validation_errors(&self) -> Option<Vec<ValidationStatus>> {
```

Comment:
Same API design comment as earlier. Consider changing these `Vec` responses to `Iter` so you're not binding yourself to an underlying implementation.

Reply:
This is only needed internally so I made it crate private